src/react/components/pages/predict/predictPage.tsx

```diff
@@ -131,15 +147,35 @@ export default class PredictPage extends React.Component<IPredictPageProps, IPre
     public async componentDidMount() {
         const projectId = this.props.match.params["projectId"];
         if (projectId) {
-            const project = this.props.recentProjects.find((project) => project.id === projectId);
-            await this.props.actions.loadProject(project);
-            this.props.appTitleActions.setTitle(project.name);
+            await this.loadProject(projectId);
         }
-
         document.title = strings.predict.title + " - " + strings.appName;
     }
 
-    public componentDidUpdate(prevProps, prevState) {
+    public async componentDidUpdate(prevProps, prevState) {
+        const onPredictPage = (new RegExp("predict$")).test(this.props.match.url)
+        if (!onPredictPage) {
+            return; // don't update if not on the predict page
+        }
+
+        if (!this.props.project) {
+            const projectId = this.props.match.params["projectId"];
+            if (projectId) {
+                await this.loadProject(projectId);
+            }
+        }
+
+        if (this.props.project?.predictModelId && !(this.props.project?.recentModelRecords) &&
```

Comment:
why do we need round brackets around this.props.project?.recentModelRecords and this.state.couldNotGetRecentModel